src/Rings/mpoly-localizations.jl

```diff
@@ -3018,14 +3018,24 @@ true
 ### Some auxiliary functions
 
 @attr MPolyLocalizedIdeal function radical(I::MPolyLocalizedIdeal)
-  J = pre_saturated_ideal(I)
-  return ideal(base_ring(I), gens(radical(J)))
+  has_attribute(I, :is_prime) && get_attribute(I, :is_prime) && return I
+  # heuristics have shown that the radical of the saturated ideal 
+  # is often quicker 
+  #J = pre_saturated_ideal(I)
+  J = saturated_ideal(I)
+  R = base_ring(I)
+  return ideal(base_ring(I), [g for g in R.(gens(radical(J))) if !is_zero(g)])
 end
 
-@attr function dim(I::MPolyLocalizedIdeal)
+@attr function dim(I::MPolyLocalizedIdeal{RT}) where {RT<:MPolyLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyPowersOfElement}}
   return dim(saturated_ideal(I))
 end
 
+@attr function dim(I::MPolyLocalizedIdeal{RT}) where {RT<:MPolyLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyComplementOfPrimeIdeal}}
+  return dim(saturated_ideal(I)) - dim(prime_ideal(inverted_set(base_ring(I))))
+end
+
+
```

Comment:
This breaks the tests. You also need a variant for complement of point ideal. There you can simply use dim(saturated_ideal(I)), but it would be better to pass through shifted_ideal.